scripts/duplicate_detector.py

```
@@ -0,0 +1,275 @@
+                page_issues = json.loads(response.read().decode())
+                if not page_issues:
+                    break
+                issues.extend(page_issues)
+                page += 1
+        except Exception as e:
+            logging.error('Error fetching issues on page %s: %s', page, e)
+            break
+    return issues
+
+
+def main() -> None:
```

Ruff (0.15.20)

[warning] Too many branches (17 > 12)

(PLR0912)